### PR TITLE
add typechecking and formatting workflow

### LINKED ISSUE #61: Add typechecking and formatting to CI
use tscircuit plop

/bounty $5

### PR DESCRIPTION
**Pull request by techmannih**
close #61 
/claim #61

### CODE REVIEW

**Review by imrishabh18 — CHANGES_REQUESTED** [review]
@techmannih As you can see the format check is failing. Please add the script for it in the `package.json` and the `biome` dependency if not present. You can take a look at other repos for reference

**Comment by imrishabh18** [comment]
What is the reason for the lockfile getting added?

**Review by imrishabh18 — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -13,7 +13,7 @@
     "preview": "vite preview",
     "storybook": "storybook dev -p 6006",
     "build-storybook": "storybook build",
-    "format": "biome format . --write"
+    "format:check": "biome format . --write"
```
Comment (seveibar) [inline package.json:16]: @techmannih `:check` scripts shouldn't write, make sure to reference other repos...

```suggestion
    "format": "biome format .",
    "format:check": "biome format . --write"
```

File: package.json [hunk 2]
```diff
@@ -13,7 +13,8 @@
     "preview": "vite preview",
     "storybook": "storybook dev -p 6006",
     "build-storybook": "storybook build",
-    "format": "biome format . --write"
+    "format": "biome format .",
+    "format:check": "biome format . --write"
```
Comment (seveibar) [inline package.json:17]:
```suggestion
    "format": "biome format . --write",
    "format:check": "biome format ."
```

Comment (seveibar) [reply package.json:17]: @techmannih this is the correct suggestion

**Pull request merged**